examples/algorithms/plot_subgraphs.py

```diff
@@ -17,96 +17,61 @@ def graph_partitioning(G, plotting=True):
     """Partition a directed graph into a list of subgraphs that contain
     only entirely supported or entirely unsupported nodes.
     """
-    # Determine indexes of all supported and unsupported nodes.
-    node_supported_index = []
-    node_unsupported_index = []
-    for v in G.nodes(data=True):
-        if v[1]["node_type"] == "supported":
-            node_supported_index.append(v[0])
-        if v[1]["node_type"] == "unsupported":
-            node_unsupported_index.append(v[0])
+    # Categorize nodes by their node_type attribute
+    supported_nodes = {n for n, d in G.nodes(data="node_type") if d == "supported"}
+    unsupported_nodes = {n for n, d in G.nodes(data="node_type") if d == "unsupported"}
```

Comment:
This is a bit inefficient since it loops through all the nodes twice. It could be replaced with something like:

```suggestion
    supported_nodes, unsupported_nodes = set(), set()
    for node, ntype in G.nodes(data="node_type"):
        if ntype == "supported":
            supported_nodes.add(node)
        elif ntype == "unsupported":
            unsupported_nodes.add(node)
```

I personally find set comprehensions more readable, but am happy either way!

Reply:
I prefer the set comprehensions even though they involve looping through the nodes twice. But it's not a strong preference.